ambry-store/src/main/java/com/github/ambry/store/CompactionPolicy.java

```diff
@@ -37,6 +37,6 @@
    * @throws StoreException
    */
   CompactionDetails getCompactionDetails(long totalCapacity, long usedCapacity, long segmentCapacity,
-      long segmentHeaderSize, List<String> logSegmentsNotInJournal, BlobStoreStats blobStoreStats)
+      long segmentHeaderSize, List<String> logSegmentsNotInJournal, BlobStoreStats blobStoreStats, CompactionPolicyCounter compactionPolicyCounter)
```

Comment:
I was wonder if you would consider this way. Using BlobStoreStats to pass store id, which is essentially the same as replica id number, to CompactionPolicy. And in CompactionPolicy, we keep a map from store id to a counter? In this way, we don't have to change interface here.

Reply:
Updated.